blog/content/post/2024-08-19-secureboot-freebsd.md

```diff
@@ -0,0 +1,397 @@
+
+### Launching QEMU with an emulated filesystem
+
+Create a directory for the QEMU firmware files, and a directory for the test
+EFI files:
+
+```bash
+mkdir -p ~/qemu_test/efi
+```
+
+Download `OVMF_CODE_RELEASE.fd` and `OVMF_VARS_RELEASE.fd` and place them in
+the `~/qemu_test` directory. Place the `signed-loader.efi` and `.cer` files in
+the `~/qemu_test/efi` directory:
+
+```bash
+cp ~/signed-loader.efi ~/qemu_test/efi/
+cp ~/testcert.cer ~/qemu_test/efi/
+cd ~/qemu_test
+```
+
+Run Dasharo firmware in QEMU, mounting the EFI directory as a virtual fat
+drive:
+
+```bash
+qemu-system-x86_64 -machine q35,smm=on \
+    -m 1G \
+    -global driver=cfi.pflash01,property=secure,value=on \
+    -drive if=pflash,format=raw,unit=0,file=OVMF_CODE_RELEASE.fd,readonly=on \
+    -drive if=pflash,format=raw,unit=1,file=OVMF_VARS_RELEASE.fd \
+    -global ICH9-LPC.disable_s3=1 \
+    -drive file=fat:rw:efi
+```
+
+### Enrolling the certificate
+
+#### Preparing clean Secure Boot
+
+1. Press `F2` to enter setup
+    ![setup menu](/img/setup_root.png)
+
+1. Navigate to `Device Manager/Secure Boot Configuration`
+    ![secure boot menu](/img/sboot_init.png)
+
+1. Enable `Custom Mode`
+    ![secure boot custom mode](/img/custom_mode.png)
+
+1. Enter the new `Advanced Secure Boot Keys Management` menu
+    ![advanced secure boot keys management](/img/advanced_sboot.png)
+
+1. Select `Reset to default Secure Boot Keys`
+    ![reset to default sboot keys](/img/reset_to_def.png)
+
+1. Select `YES`
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Reset the platform by pressing `ESC` twice and selecting `Reset`
+
+#### Enrolling the ceritficate
+
+1. Press `F2` to enter setup again
+    ![setup menu](/img/setup_root.png)
+
+1. Navigate to `Device Manager/Secure Boot Configuration`
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Enable `Custom Mode` and enter the `Advanced Secure Boot Keys Management`
+    menu again.
+    ![advanced secure boot keys management](/img/advanced_sboot.png)
+
+1. Navigate to `DB Options/Enroll Signature/Enroll Signature Using File/`.
+    If the EFI directory is properly mounted, you should see a single entry in
+    this menu, similar to this
+     ![efi directory entry](/img/efi_entry.png)
+
+1. Select the entry, and then select `testcert.cer`
+     ![testcert.cer](/img/testcert.png)
+
+1. Select `Commit Changes and Exit`
+     ![commit changes and exit](/img/commit_changes.png)
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Make sure that `Current Secure Boot State` is enabled
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Reset the platform again
+
+### Booting FreeBSD
+
+1. Wait for Dasharo to boot into UEFI Shell
+    ![EFI shell](/img/efi_shell.png)
+
+1. Enter the filesystem and boot the signed-loader.efi
+
+  ```sh
+  fs0:
+  signed-loader.efi
+  ```
+
+  ![EFI shell commands](/img/ush_commands.png)
```

Comment:
This is part that I don't like. UEFI Shell is considered to be active debug code in use. It is essentially a lousy practice and proves nothing about UEFI Secure Boot because of its vulnerability to which it may lead (essentially breaking the chain of trust using the `mm` tool). The proof here has to be made on OVMF, which has no UEFI Shell.

Reply:
[Replaced](https://github.com/3mdeb/news-and-ideas/pull/603/commits/469cc84541acf2428e46af01de19cf63b5bb0466#diff-1e142bc7519e97361fc0d3867abcb7fa3d257a4b5ccb79795e5e00acebf7d22fR326) with instructions to add a boot option instead